types/time.go

```diff
@@ -392,19 +392,19 @@ func (t Time) CompareString(sc *stmtctx.StatementContext, str string) (int, erro
 }
 
 // roundTime rounds the time value according to digits count specified by fsp.
-func roundTime(t gotime.Time, fsp int) gotime.Time {
-	d := gotime.Duration(math.Pow10(9 - fsp))
+func roundTime(t gotime.Time, fsp int8) gotime.Time {
+	d := gotime.Duration(math.Pow10(9 - int(fsp)))
 	return t.Round(d)
 }
 
 // RoundFrac rounds the fraction part of a time-type value according to `fsp`.
-func (t Time) RoundFrac(sc *stmtctx.StatementContext, fsp int) (Time, error) {
+func (t Time) RoundFrac(sc *stmtctx.StatementContext, fsp int8) (Time, error) {
 	if t.Type == mysql.TypeDate || t.IsZero() {
 		// date type has no fsp
 		return t, nil
 	}
 
-	fsp, err := CheckFsp(fsp)
+	fsp, err := CheckFsp(int(fsp))
```

Comment:
Ditto, `CheckFsp` is better to have `int8` parameter.

Reply:
I keep it as int there to make the [test function](https://github.com/pingcap/tidb/blob/master/types/fsp_test.go#L26) behaviors like before.